### PR TITLE
Schedulers last execution error

### PR DESCRIPTION
**Pull request by mmadariaga**
Persist and show in admin portal any error happened during call csv scheduler execution

### CODE REVIEW

File: web/admin/application/library/IvozProvider/Klear/Ghost/SchedulerSuccess.php [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+    /** @var \Ivoz\Provider\Domain\Model\CallCsvScheduler\CallCsvSchedulerDto $model */
+    public function getCallCsvSchedulerIcon($model)
+    {
+        $errorMsg = $model->getLastExecutionError();
+        $lastExecution = $model->getLastExecution();
+        $lastExecutionStr = $lastExecution
+            ? $lastExecution->format('Y-m-d H:i:s')
+            : '';
+
```
Comment (Kaian) [inline web/admin/application/library/IvozProvider/Klear/Ghost/SchedulerSuccess.php]: You could return empty string response if $lastExecutionStr is empty (for the first time running the scheduler)

File: web/admin/application/library/IvozProvider/Klear/Ghost/SchedulerSuccess.php [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        // Draw a green tick if found
+        if (empty($errorMsg)) {
+
+            $response =
+                $lastExecutionStr
+                . ' <span class="ui-silk inline ui-silk-tick" title="'
+                . $errorMsg
```
Comment (Kaian) [inline web/admin/application/library/IvozProvider/Klear/Ghost/SchedulerSuccess.php]: $errorMsg is empty so you can remove it from here

File: library/Ivoz/Provider/Domain/Service/CallCsvReport/CreateByScheduler.php [hunk 3]
```diff
@@ -112,9 +113,31 @@ private function updateLastExecutionDate(CallCsvSchedulerInterface $scheduler)
...
         );
+    }
+
+    /**
+     * @param CallCsvSchedulerInterface $scheduler
+     * @param $error
+     */
+    private function setExecutionError(CallCsvSchedulerInterface $scheduler, string $error)
+    {
+        /** @var CallCsvSchedulerDto $invoiceSchedulerDto */
```
[the remark this answers is not in the document]
Comment (cruzccl) [reply library/Ivoz/Provider/Domain/Service/CallCsvReport/CreateByScheduler.php]: done

File: web/admin/application/library/IvozProvider/Klear/Ghost/SchedulerSuccess.php [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+        $lastExecution = $model->getLastExecution();
+
+        return $this->getlastExecutionReport(
+            $lastExecution,
+            $errorMsg
+        );
+    }
+
+    /** @var \Ivoz\Provider\Domain\Model\InvoiceScheduler\InvoiceSchedulerDto $model */
+    public function getInvoicechedulerLastExecutionReport($model)
```
Comment (Kaian) [inline web/admin/application/library/IvozProvider/Klear/Ghost/SchedulerSuccess.php]: getInvoice*S*chedulerLastExecutionReport

File: web/admin/application/configs/klear/model/InvoiceSchedulers.yaml [hunk 5]
```diff
@@ -41,13 +41,11 @@ production:
...
-      required: true
+      type: ghost
+      dirty: true
       source:
-        control: datetime
-        settings:
-          disabled: 'false'
-      readonly: true
+        class: IvozProvider_Klear_Ghost_SchedulerSuccess
+        method: getInvoicechedulerLastExecutionReport
```
Comment (Kaian) [inline web/admin/application/configs/klear/model/InvoiceSchedulers.yaml]: getInvoiceSchedulerLastExecutionReport

**Pull request merged**